### PR TITLE
feat: implement OpenTelemetry traces hook

### PR DESCRIPTION
**Pull request by federicobond**
## This PR

- Implements an OpenTelemetry tracing hook according to the semantic conventions for feature flags.

### Related Issues

Implements [#7](https://github.com/open-feature/python-sdk-contrib/issues/7)

### CODE REVIEW

File: openfeature/contrib/hooks/otel.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+OTEL_EVENT_NAME = "feature_flag"
+
+
+class EventAttributes:
+    FLAG_KEY = f"{OTEL_EVENT_NAME}.key"
+    FLAG_VARIANT = f"{OTEL_EVENT_NAME}.variant"
+    PROVIDER_NAME = f"{OTEL_EVENT_NAME}.provider_name"
+
+
+class OTelTracesHook(Hook):
```
Comment (beeme1mr) [inline openfeature/contrib/hooks/otel.py:17]:
```suggestion
class TracingHook(Hook):
```

This is optional but would bring it in line with the [JS Hook](https://github.com/open-feature/js-sdk-contrib/blob/main/libs/hooks/open-telemetry/src/lib/traces/tracing-hook.ts).

**Review by beeme1mr — COMMENTED** [review]
Sorry, I missed this one. Please @ mention me if I miss a review in the future.

Could you please also add a short readme that describes how to use this hook? The JS doc may be a useful reference.

https://github.com/open-feature/js-sdk-contrib/tree/main/libs/hooks/open-telemetry

**Comment by federicobond** [comment]
@beeme1mr no worries, it was only recently that I moved it from draft and added you as a reviewer.

**Comment by federicobond** [comment]
I just added a readme entry for the hook and updated its name following your recommendation.

File: openfeature/contrib/hooks/README.md [hunk 2]
```diff
@@ -1,4 +1,37 @@
 # OpenFeature Python Hooks
-Hooks are a mechanism whereby application developers can add arbitrary behavior to flag evaluation. 
-They operate similarly to middleware in many web frameworks. Please see the 
-[spec]("https://github.com/open-feature/spec/blob/main/specification/sections/04-hooks.md") for more details.
\ No newline at end of file
+Hooks are a mechanism whereby application developers can add arbitrary behavior to flag evaluation.
+They operate similarly to middleware in many web frameworks. Please see the
+[spec]("https://github.com/open-feature/spec/blob/main/specification/sections/04-hooks.md") for more details.
```
Comment (beeme1mr) [inline openfeature/contrib/hooks/README.md:4]:
```suggestion
[spec](https://openfeature.dev/specification/sections/hooks) for more details.
```

**Pull request merged**